util/staking.js

```diff
@@ -22,4 +29,86 @@ export const useAccountServiceInfo = ({ account }) =>  {
   })
 
   return { data, isLoading };
+}
+
+export const useStakingDetails = (serviceId, stakingInstance) =>  {
+  const { data, isLoading } = useQuery({
+    queryKey: ['checkpoints'],
+    enabled: !!serviceId,
+    queryFn: async () => await request(STAKING_CONTRACTS_BASE_SUBGRAPH_URL, gql`
+      {
+        checkpoints(orderBy: epoch, orderDirection: desc) {
+          id
+          epoch
+          contractAddress
+          rewards
+          blockTimestamp
+          serviceIds
+        }
+      }
+    `),
+  });
+
+  const { data: livenessPeriod, isLoading: isLivenessPeriodLoading } = useReadContract({
+    address: stakingInstance,
+    abi: STAKING_TOKEN_ABI,
+    chainId: base.id,
+    functionName: 'livenessPeriod',
```

Comment:
@kupermind pls review this on-chain reading, I use this below in 97 and 99 lines to calculate current epoch end and rewards per period. The stakingInstance address goes from `mapAccountServiceInfo`